GsaGH/Parameters/5_Results/LineResultGoo.cs

```diff
@@ -13,49 +13,50 @@ namespace GsaGH.Parameters
   /// </summary>
   public class LineResultGoo : GH_GeometricGoo<Line>, IGH_PreviewData
   {
-    public LineResultGoo(Line line, IQuantity result1, IQuantity result2, Color color1, Color color2, float size1, float size2) : base(line)
+    public int ElementId;
+    public IQuantity Result1;
+    public IQuantity Result2;
+    internal List<Line> resultLineSegments;
+    internal List<Color> _previewResultColours;
+    internal List<int> _previewResultThk;
+    private float _size1;
+    private float _size2;
+    private Color _color1;
+    private Color _color2;
+
+    public LineResultGoo(Line line, IQuantity result1, IQuantity result2, Color color1, Color color2, float size1, float size2, int id) : base(line)
     {
+      ElementId = id;
       Result1 = result1;
       Result2 = result2;
       _size1 = size1;
       _size2 = size2;
       _color1 = color1;
       _color2 = color2;
 
-      var gHGradient = new Grasshopper.GUI.Gradient.GH_Gradient();
-      gHGradient.AddGrip(0, _color1);
-      gHGradient.AddGrip(1, _color2);
-      PreviewResultSegments = new List<Line>();
-      PreviewResultColors = new List<Color>();
-      PreviewResultThk = new List<int>();
+      Grasshopper.GUI.Gradient.GH_Gradient gH_Gradient = new Grasshopper.GUI.Gradient.GH_Gradient();
```

Comment:
here use var :)

Reply:
can we do that in the refactoring you are doing?